models/image.py

```diff
@@ -635,6 +642,69 @@ def from_exposure(cls, exposure, section_id):
 
         return new
 
+    @classmethod
+    def copy_image(cls, image):
+        """Make a new Image object with the same data as an existing Image object.
+
+        This new object does not have a provenance or any relationships to other objects.
+        It should be used only as a working copy, not to be saved back into the database.
+        The filepath should also be set to a new (unique) value so as not to overwrite the original.
+        """
+        copy_attributes = [
+            'data',
+            'weight',
+            'flags',
+            'score',
+            'background',
+            'header',
+
+        ]
+        simple_attributes = [
+            'ra',
+            'dec',
+            'mjd',
+            'end_mjd',
+            'exp_time',
+            'instrument',
+            'telescope',
+            'filter',
+            'section_id',
+            'project',
+            'target',
+            'preproc_bitflag',
+            'astro_cal_done',
+            'sky_sub_done',
+            'fwhm_estimate',
+            'zero_point_estimate',
+            'lim_mag_estimate',
+            'bkg_mean_estimate',
+            'bkg_rms_estimate',
+            'ref_image_index',
+            'new_image_index',
+            'is_coadd',
+            'is_sub',
+            '_bitflag',
+            '_upstream_bitflag',
+            '_format',
+            '_type',
+        ]
```

Comment:
Probably add a "not copied attributes" array here, and then raise an exception if there are any attributes left over after copy_, simple_, and not_copied_.  (Reason: if we add/remove fields to the Image model later, we also have to edit this function.  Without something that raises an exception in this function, we could easily not remember to do that, and then this function will silently be doing the wrong thing without us realizing it.)

Reply:
I'm not sure how to do that. I don't want to simply go over all attributes in `__dict__` so I don't know if there's an automated way to verify we are copying all the stuff we mean to copy. I will add an issue to deal with this.